src/composables/accounts.ts

```diff
@@ -222,12 +248,81 @@ export function useAccounts() {
     setActiveAccountByProtocolAndIdx(protocol, account?.idx || 0);
   }
 
-  function setMnemonic(newMnemonic: string) {
+  /**
+   * Setting/Resetting the password key logs the user in/out.
+   */
+  function setEncryptionKey(newEncryptionKey: CryptoKey | null) {
+    encryptionKey.value = newEncryptionKey;
+    if (IS_EXTENSION) {
+      if (newEncryptionKey) {
+        startSession(newEncryptionKey, secureLoginTimeout.value);
+      } else {
+        endSession();
+      }
+    }
+  }
+
+  async function getEncryptionKey() {
+    return watchUntilTruthy(encryptionKey);
+  }
+
+  async function openLoginModal() {
```

Comment:
As this is opening the password login modal I suggest to rename this function to `openPasswordLoginModal`.